### PR TITLE
Use px CSS unit instead of pt

### LINKED ISSUE #6840: Use px as the default font size unit of measurement
Currently we use `pt` almost exclusively. However, `pt` is quite meaningless in the context of display on a screen. Given that everything else in bokeh in screen coordinates is given in pixels, I think we should apply the same to font sizes.

### PR DESCRIPTION
**Pull request by mattpap**
This was extracted from PR #9624. This should have been done for 2.0, however it should be mostly a non-breaking change. The problem is that pt to px conversion results in fractional numbers. We primarily use 10pt and 12pt as the defaults, for which we get:
```
10 pt -> 13.281472327365 px -> 13 px
12 pt -> 15.937766792839 px -> 16 px
```
However, where it matters (e.g. layout) we use rounded coordinates and sizes anyway, so there should be no difference after this change. Users' code using non-default values won't be affected, because we don't support unit-less values.

fixes #6840

### CODE REVIEW

File: bokeh/core/property_mixins.py [hunk 1]
```diff
@@ -178,9 +178,9 @@ class _BaseLineProps(HasProps):
 
 class _BaseTextProps(HasProps):
 
-    text_font = String("helvetica", help="""
+    text_font = String("Helvetica", help="""
```
Comment (bryevdv) [inline bokeh/core/property_mixins.py]: Why is this change necessary?

**Comment by bryevdv** [comment]
Tangentially, did we ever establish whether relative units like `rem` function on canvas?

**Comment by mattpap** [comment]
Yes, in https://github.com/bokeh/bokeh/issues/9407.

File: bokeh/core/property_mixins.py (hunk 1 above)
Comment (mattpap) [reply bokeh/core/property_mixins.py]: Not necessary, but currently we have a mixture of lower and upper case versions in stylesheets and defaults. CSS font face names are case insensitive, so it can be either. I changed it for the sake of consistency.

**Comment by bryevdv** [comment]
If we are going to make things consistent, my preference is lower case throughout for font names.

File: bokeh/core/property_mixins.py (hunk 1 above)
Comment (bryevdv) [reply bokeh/core/property_mixins.py]: If we are going to make things consistent, my preference is lower case throughout for font names.
